feather/protocol/src/packets/server/play.rs

```diff
@@ -1179,6 +1181,163 @@ packets! {
     }
 }
 
+def_enum! {
+    OperationKind (u8) {
+        0 = AddSubtract,
+        1 = AddSubtractPercent,
+        2 = MultiplyPercent
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum EntityAttributeKind {
+    MaxHealth,
+    FollowRange,
+    KnockbackResistance,
+    MovementSpeed,
+    AttackDamage,
+    AttackSpeed,
+    AttackKnockback,
+    FlyingSpeed,
+    Armor,
+    ArmorToughness,
+    Luck,
+    HorseJumpStrength,
+    ZombieSpawnReinforcements,
+}
+
+impl EntityAttributeKind {
+    pub fn from_key(key: &str) -> anyhow::Result<Self> {
+        Ok(match key {
+            "generic:max_health" => EntityAttributeKind::MaxHealth,
+            "generic:follow_range" => EntityAttributeKind::FollowRange,
+            "generic:knockback_resistance" => EntityAttributeKind::KnockbackResistance,
+            "generic:movement_speed" => EntityAttributeKind::MovementSpeed,
+            "generic:attack_damage" => EntityAttributeKind::AttackDamage,
+            "generic:attack_speed" => EntityAttributeKind::AttackSpeed,
+            "generic:attack_knockback" => EntityAttributeKind::AttackKnockback,
+            "generic:flying_speed" => EntityAttributeKind::FlyingSpeed,
+            "generic:armor" => EntityAttributeKind::Armor,
+            "generic:armor_toughness" => EntityAttributeKind::ArmorToughness,
+            "generic:luck" => EntityAttributeKind::Luck,
+            "horse:jump_strength" => EntityAttributeKind::HorseJumpStrength,
+            "zombie:spawn_reinforcements" => EntityAttributeKind::ZombieSpawnReinforcements,
+            _ => return Err(anyhow::anyhow!("invalid entity property key: '{}'", key)),
+        })
+    }
+
+    pub fn id(&self) -> String {
+        String::from(match self {
+            EntityAttributeKind::MaxHealth => "generic:max_health",
+            EntityAttributeKind::FollowRange => "generic:follow_range",
+            EntityAttributeKind::KnockbackResistance => "generic:knockback_resistance",
+            EntityAttributeKind::MovementSpeed => "generic:movement_speed",
+            EntityAttributeKind::AttackDamage => "generic:attack_damage",
+            EntityAttributeKind::AttackSpeed => "generic:attack_speed",
+            EntityAttributeKind::AttackKnockback => "generic:attack_knockback",
+            EntityAttributeKind::FlyingSpeed => "generic:flying_speed",
+            EntityAttributeKind::Armor => "generic:armor",
+            EntityAttributeKind::ArmorToughness => "generic:armor_toughness",
+            EntityAttributeKind::Luck => "generic:luck",
+            EntityAttributeKind::HorseJumpStrength => "horse:jump_strength",
+            EntityAttributeKind::ZombieSpawnReinforcements => "zombie:spawn_reinforcements",
```

Comment:
According to wiki.vg all of these identifiers are separated by `.` not by `:`. E.g. `generic.max_health` instead of `generic:max_health`.

Reply:
In https://wiki.vg/Protocol#Statistics , it mentions that they separate with `.` but it's actually `:`. I'm going with that assumption here as well, but I'll test it later.